validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/ValidatorApiHandler.java

```diff
@@ -101,7 +107,20 @@ public ValidatorApiHandler(
         .getStateAtSlot(slot)
         .thenApply(
             optionalState ->
-                optionalState.map(state -> getValidatorDutiesFromState(state, epoch, publicKeys)));
+                optionalState
+                    .map(state -> processSlots(state, slot))
```

Comment:
It might make more sense to move slot processing into `combinedChainDataClient`.

Reply:
To even recentChainData actually. I need the same exact functionality for Proto Array Fork Choice.